lib/daru/vector.rb

```diff
@@ -199,19 +199,62 @@ def initialize source, opts={}
     #   # For vectors employing hierarchial multi index
     #
     def [](*input_indexes)
-      # Get a proper index object
-      indexes = @index[*input_indexes]
+      # Get array of positions indexes
+      positions = @index.pos(*input_indexes)
```

Comment:
It's the same `MultiIndex`issue we are facing when replacing `#subset` with `#at`.